script.js

```diff
@@ -1,4 +1,89 @@
+//two game modes
+var gameModeRollDice = "Game mode roll dice";
+var gameModeDiceSequence = "Game mode dice sequence";
+
+var gameMode = gameModeRollDice;
+
+// Player 1 to start the game
+var currPlayer = "player1";
+
+// There are two players in the game
+var diceArrayPlayer1 = [];
+var diceArrayPlayer2 = [];
+
+// Each Player Total Scores
+var player1Total = 0;
+var player2Total = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  if (gameMode == gameModeRollDice && currPlayer == "player1") {
+    var dice1 = rollDice();
+    diceArrayPlayer1.push(dice1);
+    var dice2 = rollDice();
+    diceArrayPlayer1.push(dice2);
+    myOutputValue =
+      "You rolled" +
+      dice1 +
+      dice2 +
+      "Player 1, Please choose your sequence by inputting 0 (1st dice & 2nd dice) or 1 (2nd dice & 1st dice)";
+    gameMode = gameModeDiceSequence;
+    return myOutputValue;
+  } else if (gameMode == gameModeDiceSequence && currPlayer == "player1") {
+    gameMode = gameModeRollDice;
+    currPlayer = "player2";
+    if (input == 0) {
+      player1Total = diceArrayPlayer1[0] + "" + diceArrayPlayer1[1];
+      return player1Total;
+    }
+    if (input == 1) {
+      player1Total = diceArrayPlayer1[1] + "" + diceArrayPlayer1[0];
+      return player1Total;
+    }
+  }
+  if (gameMode == gameModeRollDice && currPlayer == "player2") {
+    var dice1 = rollDice();
+    diceArrayPlayer2.push(dice1);
+    var dice2 = rollDice();
+    diceArrayPlayer2.push(dice2);
+    myOutPutValue =
+      "You rolled" +
+      dice1 +
+      dice2 +
+      "Player 2, Please choose your sequence by inputting 0 (1st dice & 2nd dice) or 1 (2nd dice & 1st dice)";
+    gameMode = gameModeDiceSequence;
+    return myOutPutValue;
+  } else if (gameMode == gameModeDiceSequence && currPlayer == "player2") {
+    currPlayer = "player1";
+    gameMode = gameModeRollDice;
+    if (input == 0) {
+      player2Total = diceArrayPlayer2[0] + "" + diceArrayPlayer2[1];
+    }
+    if (input == 1) {
+      player2Total = diceArrayPlayer2[1] + "" + diceArrayPlayer2[0];
+    }
+    myOutPutValue = winningCondition();
+    return myOutPutValue;
+  }
+};
```

Comment:
I like how you are controlling the flow of the program very logically seen in the conditions you are using. It's great work Jeslin!